lib/services/role.go

```diff
@@ -2079,6 +2102,33 @@ func (set RoleSet) EnhancedRecordingSet() map[string]bool {
 	return m
 }
 
+// HostUsers returns host user information matching a server or nil if
+// a role disallows host user creation
+func (set RoleSet) HostUsers(s types.Server) *HostUsersInfo {
+	var groups []string
+	serverLabels := s.GetAllLabels()
+	for _, role := range set {
+		result, _, err := MatchLabels(role.GetNodeLabels(types.Allow), serverLabels)
+		if err != nil {
```

Comment:
If anything goes wrong in this function, it will be very hard to troubleshoot since it swallows all errors. I would make the following changes:

- Update the `HostUsers` method to return error as a second return parameter.
- Do not swallow errors here, instead return them to the caller and let them log/handle them.